go/internal/feast/onlinestore/cassandraonlinestore.go

```diff
@@ -0,0 +1,392 @@
+	for _, fvName := range featureViewNames {
+		uniqueNames[fvName] = 0
+	}
+	if len(uniqueNames) != 1 {
+		return nil, fmt.Errorf("rejecting OnlineRead as more than 1 feature view was tried to be read at once")
+	}
+
+	serializedEntityKeys, serializedEntityKeyToIndex, err := c.buildCassandraEntityKeys(entityKeys)
+
+	if err != nil {
+		return nil, fmt.Errorf("error when serializing entity keys for Cassandra")
+	}
+	results := make([][]FeatureData, len(entityKeys))
+	for i := range results {
+		results[i] = make([]FeatureData, len(featureNames))
+	}
+
+	featureNamesToIdx := make(map[string]int)
+	for idx, name := range featureNames {
+		featureNamesToIdx[name] = idx
+	}
+
+	featureViewName := featureViewNames[0]
+
+	// Prepare the query
+	tableName := c.getFqTableName(featureViewName)
+	cqlStatement := c.getCQLStatement(tableName, featureNames)
+
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(len(serializedEntityKeys))
+
+	errorsChannel := make(chan error, len(serializedEntityKeys))
+	for _, serializedEntityKey := range serializedEntityKeys {
+		go func(serEntityKey any) {
+			defer waitGroup.Done()
+
+			iter := c.session.Query(cqlStatement, serEntityKey).WithContext(ctx).Iter()
+
+			rowIdx := serializedEntityKeyToIndex[serializedEntityKey.(string)]
+
+			// fill the row with nulls if not found
+			if iter.NumRows() == 0 {
+				for _, featName := range featureNames {
+					results[rowIdx][featureNamesToIdx[featName]] = FeatureData{
+						Reference: serving.FeatureReferenceV2{
+							FeatureViewName: featureViewName,
+							FeatureName:     featName,
+						},
+						Value: types.Value{
+							Val: &types.Value_NullVal{
+								NullVal: types.Null_NULL,
+							},
+						},
+					}
+				}
+				return
+			}
+
+			scanner := iter.Scanner()
+			var entityKey string
+			var featureName string
+			var eventTs time.Time
+			var valueStr []byte
+			var deserializedValue types.Value
+			rowFeatures := make(map[string]FeatureData)
+			for scanner.Next() {
+				err := scanner.Scan(&entityKey, &featureName, &eventTs, &valueStr)
+				if err != nil {
+					errorsChannel <- errors.New("could not read row in query for (entity key, feature name, value, event ts)")
+					return
+				}
+				if err := proto.Unmarshal(valueStr, &deserializedValue); err != nil {
+					errorsChannel <- errors.New("error converting parsed Cassandra Value to types.Value")
+					return
+				}
+
+				if deserializedValue.Val != nil {
+					// Convert the value to a FeatureData struct
+					rowFeatures[featureName] = FeatureData{
+						Reference: serving.FeatureReferenceV2{
+							FeatureViewName: featureViewName,
+							FeatureName:     featureName,
+						},
+						Timestamp: timestamppb.Timestamp{Seconds: eventTs.Unix(), Nanos: int32(eventTs.Nanosecond())},
+						Value: types.Value{
+							Val: deserializedValue.Val,
+						},
+					}
+				} else {
+					// Return FeatureData with a null value
+					rowFeatures[featureName] = FeatureData{
```

Comment:
Seems like this piece of code is repeating, would be nicer to call a helper function instead, something like :

func buildNullFeatureData(featureName, featureViewName string) FeatureData {
    return FeatureData{
        Reference: serving.FeatureReferenceV2{
            FeatureViewName: featureViewName,
            FeatureName:     featureName,
        },
        Value: types.Value{
            Val: &types.Value_NullVal{NullVal: types.Null_NULL},
        },
    }
}

Reply:
Removed this else block entirely, if we don't put this `FeatureData` in the `rowFeatures` the final loop of the goroutine will fill the slot with Null